src/graph.cpp

```diff
@@ -81,40 +81,47 @@ bool Graph::addBidirectionalEdge(const std::string &source, const std::string &d
 }
 
 /**
- * Standard Edmonds-Karp algorithm to find the the network's max flow
+ * Single-source or Multi-source Edmonds-Karp algorithm to find the the network's max flow
  * Time Complexity: O(|VE²|)
- * @param source - Id of the source Vertex
+ * @param source - List of ids of the source Vertex(es)
  * @param target - Id of the target Vertex
  */
-void Graph::edmondsKarp(const std::string &source, const std::string &target) {
+unsigned int Graph::edmondsKarp(const std::list<std::string> &source, const std::string &target) {
     for (Vertex *v: vertexSet) {
         for (Edge *e: v->getAdj()) {
             e->setFlow(0);
         }
     }
 
     while (path(source, target)) {
-        augmentPath(source, target, findBottleneck(source, target));
+        Vertex *it_source = findVertex(target)->getPath()->getOrig();
```

Comment:
What is the purpose of this code?

Reply:
The augmentPath() method only takes 1 station as source. In a multi-source scenario for the Edmonds-Karp, we need to find what source is the start of the path we are augmenting. Not sure if I'm making myself clear.

Reply:
The path function is only ever gonna find one path, as you can see by it returning as soon as it find a path to target. What the multisource bfs entails is more options for that path, but the bfs will only ever choose one (the shortest)

Reply:
Then we should remove the "source" parameter from the augmentPath(), that was already implemented when I started working... cof

Reply:
I ran some tests and I think that would work, here is the code I propose instead then:

```
double Graph::findBottleneck(const std::string &target) const {
    Vertex *currentVertex = findVertex(target);
    double currBottleneck;
    double bottleneck = INF;

    while (currentVertex->getPath() != nullptr) {

        if (currentVertex->getPath()->getDest() == currentVertex) //Regular Edge
        {
            currBottleneck = currentVertex->getPath()->getCapacity() -
                             *currentVertex->getPath()->getFlow();
            currentVertex = currentVertex->getPath()->getOrig();
        } else //Residual Edge (was traversed backwards)
        {
            currBottleneck = *currentVertex->getPath()->getFlow();
            currentVertex = currentVertex->getPath()->getDest();
        }

        if (currBottleneck < bottleneck)
            bottleneck = currBottleneck;

    }
    return bottleneck;
}
```



```
void Graph::augmentPath(const std::string &target, const unsigned int &value) const {
    Vertex *currentVertex = findVertex(target);

    while (currentVertex->getPath() != nullptr) {
        Edge *currentPath = currentVertex->getPath();

        if (currentPath->getDest() == currentVertex) //Regular Edge
        {
            currentPath->setFlowValue(*(currentPath->getFlow()) + value);
            currentVertex = currentVertex->getPath()->getOrig();

        } else { //Residual Edge (was traversed backwards)
            currentPath->setFlowValue(*(currentPath->getFlow()) - value);
            currentVertex = currentVertex->getPath()->getDest();
        }
    }
}
```

Add it to your branch if you agree